packages/lib-user/src/components/Contributors/helpers/handleGenerateExport.js

```diff
@@ -0,0 +1,66 @@
+import asyncStates from '@zooniverse/async-states'
+import { arrayOf, func, number, shape, string } from 'prop-types'
+
+import { getAllUsers } from './getAllUsers'
+import { generateExport } from './generateExport'
+
+export async function handleGenerateExport({
+  group,
+  memberIdsPerStats,
+  projects,
+  setDownloadUrl,
+  setExportProgress,
+  setExportStatus,
+  setErrorMessage,
+  stats
+}) {
+  setExportStatus(asyncStates.loading)
+  setExportProgress(0)
+
+  try {
+    const allUsers = await getAllUsers({ memberIdsPerStats, setExportProgress })
+
+    const { filename, dataExportUrl } = await generateExport({
```

Comment:
I'm not sure without doing some research, but I presume there's way to calculate the actual size of the dataExport once it's generated. If so, the actual size could be added to the success state component.